runtime/src/iree/hal/local/executable_library.h

```diff
@@ -408,6 +408,13 @@ typedef struct iree_hal_executable_export_table_v0_t {
   // verbose logging. The string values, when present, may be attached to
   // tracing/debugging events related to the entry point.
   const char* const* tags;
+
+  // Optional table of source files which src_locs index into, 1:1 with ptrs.
+  const char* const* src_files;
```

Comment:
We should make this a struct array instead - there's never a case where src_files XOR src_locs is valid and for efficiency we want to include the string lengths so we don't need to compute them on each event at runtime. Cache-wise since we always access all of the fields together having them all sit together in memory reduces the number of cache misses.

```c
typedef struct {
  uint32_t line;
  uint32_t path_length;
  const char* path;
  // could eventually have source contents embedded too
} iree_hal_executable_src_loc_t;

const iree_hal_executable_src_loc_t* src_locs;
```

Reply:
Relating to the `IREE_HAL_EXECUTABLE_LIBRARY_VERSION` comment, should that struct be named `iree_hal_executable_src_loc_v0_t`? (matching the other `_v0_` structs)